### PR TITLE
refactor(common): nuxt.config.js and dynamic server options (latest process.env)

### PR DESCRIPTION
**Pull request by galvez**
Shouldn't be needed after https://github.com/nuxt/nuxt.js/pull/4207#issuecomment-433521714

### CODE REVIEW

**Comment by galvez** [comment]
@pi0 or maybe not? will check later, gtg now :)

**Comment by aldarund** [comment]
Tests failing so not so easy :)

**Comment by pi0** [comment]
@pimlie is our cli hero :D

**Comment by pimlie** [comment]
It seems to me this could be merged after changing the test. The failing test is a direct result of the expectation that ./utils.js set a default for host/port. If we want to improve that by moving it to nuxt.config.js like all other defaults then we just cant test for that here.

**Comment by galvez** [comment]
@pimlie ha, you're totally right -- I had a feeling about this too but couldn't look into it at the time. Will update.

**Comment by galvez** [comment]
So I've stopped at trying to load a fresh version of NuxtConfig because we need to pick up fresh `process.env` values. Suggestions welcome :)

**Comment by galvez** [comment]
Back later!

**Comment by pimlie** [comment]
Why and where is the fresh config needed? Its a bit hard to follow your thought process with all the fix descriptions, but if you are trying to fix the failing test which checks `env-host` then you will never succeed in that ;)

These are all strict unit tests which (try) to only test one (class) method at the time (and only stuff from this package). As setting the host from env is now moved from this package to nuxt/common the `env-host` test can be removed here I guess (and it should be added to a nuxt/common test).

**Comment by pimlie** [comment]
Ah ok, thanks. If we have that problem in nuxt/common then its probably easier to just call `jest.resetModules()` or (sometimes that doesnt work) `delete require.cache[...path.to.nuxt-config]` as it seems it'd only be a test issue and not a real world issue.

**Review by pi0 — APPROVED** [review]
